### PR TITLE
ffmpegthumbnailer 2.0.10

### PR DESCRIPTION
**Pull request by L2G**
* New upstream version
* Home moved from Google Code to GitHub
* Build process changed from GNU-style configure et al. to CMake, so add new patch and inreplace to massage the build for Homebrew
* Add test
* Other changes for `brew audit --strict`

### CODE REVIEW

**Comment by L2G** [comment]
I need to fix the formula syntax to work with Ruby 1.9 so it will work in Mountain Lion.

File: Library/Formula/ffmpegthumbnailer.rb [hunk 1]
```diff
@@ -12,16 +11,52 @@ class Ffmpegthumbnailer < Formula
...
   def install
-    system "./configure", "--disable-debug", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
-    system "make install"
+    # The CMake script does not allow fine-grained control of man directories or
+    # other installation paths beneath the prefix
+    inreplace "CMakeLists.txt", "${CMAKE_INSTALL_PREFIX}/man/man1", man1
+
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args, "-DENABLE_GIO=ON",
```
Comment (dunn) [inline Library/Formula/ffmpegthumbnailer.rb]: Older versions of Ruby can't have anything in a list like this after an `*exploded` array. Just swapping `*std_cmake_args` with `"-DENABLE_GIO=ON"` should fix the ML build.

Comment (dunn) [inline Library/Formula/ffmpegthumbnailer.rb]: Woops, looks like you already know this!

File: Library/Formula/ffmpegthumbnailer.rb [hunk 2]
```diff
@@ -12,16 +11,52 @@ class Ffmpegthumbnailer < Formula
...
-  depends_on 'ffmpeg'
+  patch :DATA
 
   def install
-    system "./configure", "--disable-debug", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
-    system "make install"
+    # The CMake script does not allow fine-grained control of man directories or
+    # other installation paths beneath the prefix
+    inreplace "CMakeLists.txt", "${CMAKE_INSTALL_PREFIX}/man/man1", man1
```
Comment (dunn) [inline Library/Formula/ffmpegthumbnailer.rb]: Has this been reported upstream? They can fix their build script, right?

File: Library/Formula/ffmpegthumbnailer.rb [hunk 3]
```diff
@@ -12,16 +11,52 @@ class Ffmpegthumbnailer < Formula
...
+  depends_on "pkg-config" => :build
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "ffmpeg"
 
-  depends_on 'pkg-config' => :build
-  depends_on 'jpeg'
-  depends_on 'libpng'
-  depends_on 'ffmpeg'
+  patch :DATA
```
Comment (dunn) [inline Library/Formula/ffmpegthumbnailer.rb]: Can this be accomplished with `-DMACOSX_RPATH=OFF` as an argument to cmake?

File: Library/Formula/ffmpegthumbnailer.rb (hunk 2 above)
Comment (L2G) [inline Library/Formula/ffmpegthumbnailer.rb]: I have a feeling they used some kind of template or generator to create their CMakeLists.txt, then never looked at it again except to bump the version number. So yeah, there is a lot of room for improvement in there, this just being the tip of the iceberg (or the mouth of the rabbit hole).

**Comment by L2G** [comment]
After more careful inspection of the GitHub repo, the bzip2 tarball, and the gzip tarball, I believe the bzip2 tarball was produced by an experimental run of CPack. I don't think the CMake build is (or was intended to be) ready for prime time. I'm going to go back and try this again using the gzip tarball with Automake and Autoconf.

**Comment by dunn** [comment]
Merged in 325ab8f68bec4a5c0de1be48150e99748446fbda. Thanks for figuring that out!

**Pull request closed without merging**